.gitignore

```diff
@@ -6,3 +6,4 @@ src/run.ts
 .vscode
 .DS_Store
 scratch
+docs/
```

Comment:
If you wanted, it might be useful to check them in here, and then have a pr step that generates + checks for a diff (similar to : https://github.com/pinecone-io/auth0-config/blob/main/package.json#L4)

Especially if you plan to auto-sync these with the hosted docs, could be nice to have that extra check before anything goes public

Reply:
_Note: Sorry, this comment got super long. Basically me writing a docs one-pager strategy in a github comment 😂_ 

I'm exploring options on how to publish these, but it's looking less and less likely that these will be deployed to readme.io because the cost is so high for our use case. Something like static hosting on github pages can probably meet our needs, but need to figure out how I can build different doc versions for each versioned release. I decided not to check in this `docs` directory to main while I explore those options, but will revisit in the future if it makes sense.

It's complicated because versioned docs is closely tied to how we make releases and keep track of them. I think I need some kind of scheme where every time a release is made, we create a release branch that sticks around in case future doc changes are needed to correct documentation mistakes for previous releases. So, for example, today if I release it creates a new git tag for that version, like `v0.6.0`, but there's no long-lived release branch because so far we've only ever released from the tip of `main` and never had to deal with docs in our source. To support future maintenance of versioned docs I think I should probably also create a `release-v0.6.0` branch at release time which contains the `v0.6.0` tag but which also provides a way for version-specific docs to continue to receive updates post-release. Because important mistakes and omissions definitely happen.

If I had this kind of release branch structure, when building the static docs site, I would need a CI process that visits each `release-$VERSION` branch, builds the docs on that branch, and copies the output into a versioned subdirectory like `docs/v0.6.0`. In the end, I'd like to have a docs structure that looks something like this with all the versioned output tracked in a single branch (perhaps a branch called `docs` or `gh-pages`):

```
docs
├── index.html (intro, navigation to version-specific docs)
├── main (unreleased development)
│   ├── assets
│   ├── classes
│   ├── functions
│   ├── interfaces
│   ├── modules
│   └── types
├── v0.6.0
│   ├── assets
│   ├── classes
│   ├── functions
│   ├── interfaces
│   ├── modules
│   └── types
├── v0.7.0
│   ├── assets
│   ├── classes
│   ├── functions
│   ├── interfaces
│   ├── modules
│   └── types
└── v0.8.0
    ├── assets
    ├── classes
    ├── functions
    ├── interfaces
    ├── modules
    └── types
```

This is something that could run after every single commit to `main` or a `release-$VERSION` branch. In this type of approach we should probably check in CI that docs can compile without errors, but there's no need to check-in docs changes to main; we assume there could be changes and rebuild all the docs after every push to these branches.

Reply:
Gotcha, I hadn't really considered the need to show historical docs but definitely makes sense for versioned clients like this -- thanks for the all the info!